website/integrations: add ironclad (#14446)
goauthentik/authentik: website/integrations/services/ironclad/index.mdx

>
> -- https://ironcladapp.com/

## Preparation

The following placeholders are used in this guide:

- `authentik.company` is the FQDN of the authentik installation.

:::note
This documentation lists only the settings that you need to change from their default values. Be aware that any changes other than those explicitly mentioned in this guide could cause issues accessing your application.
:::

## authentik configuration

To support the integration of Ironclad with authentik, you need to create three property mappings, and an application/provider pair in authentik.

### Create property mappings

Ironclad requires both a first and last name for each user, but by default, authentik only provides a full name as a single string. Ironclad also requires the email attribute in a specific format. Therefore, property mappings must be created to provide the email, first name, and last name to Ironclad.

1. Log in to authentik as an administrator, and open the authentik Admin interface.
2. Navigate to **Customization** > **Property Mappings**, click **Create**, select **SAML Provider Property Mappings**, and click **Next**.
3. Configure the first mapping for the user's first name:

>     - **Name**: `firstName`

Reviewer: I'm not sure why markdown (or Docusaurus?) is adding a space between these three bullets but it would be nice to remove them and make this list a bit more compact.

Author: This is docusaurus being weird. I've tried a few changes. But as far as I'm aware, there's nothing that can be changed in the markdown to fix it.

2. Navigate to **Applications** > **Providers** and click the **Edit** icon of the newly created Push Security provider.
3. Under **Protocol settings**, set the following required configurations:
    - **ACS URL**: Set to the **Callback** URL provided by Ironclad (e.g. `https://ironcladapp.com/saml/<customer-ID>/callback`).
    - **Issuer**: Set to the **Service Provider Identifier** provided by Ironclad (e.g. `na1.ironcladapp.com`).
4. Click **Update**.

## Download the metadata file

1. Log into authentik as an administrator, and open the authentik Admin interface.
2. Navigate to **Applications** > **Providers** and click on the name of the newly created Ironclad provider.
3. Under **Metadata** click **Download**. This metadata file will be required in the next section.

## Complete Ironclad configuration

1. Return to the Ironclad SAML configuration page and under **IdP Configuration XML** click on **Choose file**. Select the metadata file that you downloaded from authentik.
2. Set **Entry point** to `https://authentik.company/application/saml/<application-slug>/sso/binding/redirect/`.
3. Click **Save**.

:::note

> Ironclad user accounts must be enabled for SSO login. The original Ironclad administrator account can only be enabled for SSO by Ironclad. You will need to request this from Ironclad support via `support@ironcladapp.com`.

Author: Rewrote this line. A hybrid of yours and mine.

I don't like "Ironclad user accounts must be enabled to support SSO login" because it could be interpreted as the account being disabled/enabled, rather than SSO being enabled/disabled.

---
title: Integrate with Ironclad
sidebar_label: Ironclad
support_level: community
---

## What is Ironclad

> Ironclad is a contract lifecycle management (CLM) platform that gives business and legal teams an easy-to-use platform with AI-powered tools to handle every aspect of the contract lifecycle.
>
> -- https://ironcladapp.com/

## Preparation

The following placeholders are used in this guide:

- `authentik.company` is the FQDN of the authentik installation.

:::note
This documentation lists only the settings that you need to change from their default values. Be aware that any changes other than those explicitly mentioned in this guide could cause issues accessing your application.
:::

## authentik configuration

To support the integration of Ironclad with authentik, you need to create three property mappings, and an application/provider pair in authentik.

### Create property mappings

> Ironclad requires a first and last name for each user. By default authentik only provides a full name, as a single string value. Ironclad also requires the email attribute in a specific form. Therefore you need to create property mappings to provide email, first names, and last names to Ironclad.

Reviewer 2: ```suggestion
Ironclad requires both a first and last name for each user, but by default, authentik only provides a full name as a single string. Additionally, Ironclad expects the email attribute in a specific format. Therefore, property mappings which supply the email, first name, and last name to Ironclad must be created.
```

1. Log in to authentik as an administrator, and open the authentik Admin interface.
2. Navigate to **Customization** > **Property Mappings** and click **Create**.

>     - **Select type**: select **SAML Provider Property Mapping** as the property mapping type.

Reviewer 2: Can we tweak this up a bit like in a PR which I forget it's ID where there were like 5 mappings to create f the same type as to avoid the repetition?

2. Navigate to **Applications** > **Applications** and click **Create with Provider** to create an application and provider pair. (Alternatively you can first create a provider separately, then create the application and connect it with the provider.)

    - **Application**: provide a descriptive name, an optional group for the type of application, the policy engine mode, and optional UI settings. Note the **slug** value because it will be required later.
    - **Choose a Provider type**: select **SAML Provider** as the provider type.
    - **Configure the Provider**: provide a name (or accept the auto-provided name), the authorization flow to use for this provider, and the following required configurations.

        - Temporarily set the **ACS URL** to `https://temp.temp`
        - Set **Service Provider Binding** to `Post`.
        - Under **Advanced protocol settings**:
            - Set an available signing certificate.
            - Toggle off **Sign assertions**.
            - Toggle on **Sign responses**.
            - **Property mappings**:
                - Click the **x** button to remove all selected property mappings.

>                 - Add the `firstName`, `lastName`, and `email` property mappings that you created in the previous section to **Selected User Property Mappings**.

Reviewer 2: ```suggestion
                - Under **Selected User Property Mappings**, add the `firstName`, `lastName`, and `email` property mappings that were created in the previous section.
```

I prefer phrasing it this way, however, we can keep it like before.

    - **Configure Bindings** _(optional)_: you can create a [binding](/docs/add-secure-apps/flows-stages/bindings/) (policy, group, or user) to manage the listing and access to applications on a user's **My applications** page.

3. Click **Submit** to save the new application and provider.

## Ironclad configuration

> 1. Login to the [Ironclad dashboard](https://ironcladapp.com/signin) as an administrator.

Reviewer 2: ```suggestion
1. Log in to the [Ironclad dashboard](https://ironcladapp.com/signin) as an administrator.
```

> 2. Click on your profile icon at the top right, and then click on **Company Settings**.

Reviewer 2: ```suggestion
2. Click your profile icon in the top-right corner, then select **Company Settings**.
```

> 3. In the sidebar, click on **Integrations**, and then under **Other Integrations** click on **SAML**.

Reviewer 2: ```suggestion
3. In the sidebar, go to **Integrations**, and under **Other Integrations**, click **SAML**.
```

> 4. Click **Add SAML Integration**, and then click on **Show Additional IdP Settings**.

Reviewer 2: ```suggestion
4. Click **Add SAML Integration**, then select **Show Additional IdP Settings**.
```

> 5. Note the **Callback** URL and **Service Provider Identifier** values because they will be needed in the next section.

Reviewer 2: ```suggestion
5. Copy the **Callback URL** and **Service Provider Identifier**, you’ll need them in the next section.
```

> 6. Leave this page open because you will need to return to it after the next section.

Reviewer 2: ```suggestion
6. Keep this page open; you’ll return to it shortly.
```

Reviewer 2: Might be a bit too short?

## Reconfigure authentik provider

1. Log in to authentik as an administrator, and open the authentik Admin interface.
2. Navigate to **Applications** > **Providers** and click the **Edit** icon of the newly created Push Security provider.
3. Under **Protocol settings**, set the following required configurations:

>     - **ACS URL**: Set to the **Callback** URL from Ironclad (e.g. `https://ironcladapp.com/saml/<customer-ID>/callback`).

Reviewer 2: ```suggestion
    - **ACS URL**: Set to the **Callback** URL provided by Ironclad (e.g. `https://ironcladapp.com/saml/<customer-ID>/callback`).
```

Reviewer 2: oh and what does the UI say? is it just Callback or Callback URL? That influences bolding

>     - **Issuer**: Set to the **Service Provider Identifier** value from Ironclad (e.g. `na1.ironcladapp.com`).

Reviewer 2: ```suggestion
    - **Issuer**: Enter the **Service Provider Identifier** from Ironclad (e.g. `na1.ironcladapp.com`).
```

> 4. Click **Update**

Reviewer 2: ```suggestion
4. Click **Update**.
```
